qiskit/quantum_info/synthesis/one_qubit_decompose.py

```diff
@@ -91,18 +91,20 @@ class OneQubitEulerDecomposer:
             :math:`R\left(\theta+\pi,\frac{\pi}{2}-\lambda\right)`
     """
 
-    def __init__(self, basis='U3'):
+    def __init__(self, basis='U3', u3_only=True):
```

Comment:
Instead of adding a kwarg specific to the `u3` case, could this be its own entry in `ONE_QUBIT_EULER_BASIS_GATES` ? Having separate methods for`_circuit_u3` and `_circuit_u321` is also reasonable.

Reply:
Updated for U321 in https://github.com/Qiskit/qiskit-terra/pull/5554/commits/d06620e5a09f706a5ab6db8020a3b1a967c11eb5